pkg/operator/ceph/cluster/controller.go

```diff
@@ -555,3 +562,26 @@ func (c *ClusterController) deleteOSDEncryptionKeyFromKMS(currentCluster *cephv1
 
 	return nil
 }
+
+// updateStatus updates an object with a given status
+func (r *ReconcileCephCluster) updateObservedGenerationInStatus(namespacedName types.NamespacedName, observedGeneration int64) {
+	// Fetch the cephCluster instance
+	cephCluster := &cephv1.CephCluster{}
+	err := r.client.Get(r.opManagerContext, namespacedName, cephCluster)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("cephCluster resource not found. Ignoring since object must be deleted.")
+			return
+		}
+		// Error reading the object - requeue the request.
+		logger.Warningf("failed to retrieve cephCluster resource %q to update ObservedGeneration in status. %v", namespacedName, err)
+		return
+	}
+
```

Comment:
Here is it required to test if the `cephCluster.Status==nil`?
I think not because we are updating status during the reconcile at many places for cephcluster.